src/components/post/dto/create-post.dto.ts

```diff
@@ -0,0 +1,36 @@
+import { Field, InputType, ObjectType } from '@nestjs/graphql';
+import { Type } from 'class-transformer';
+import { ValidateNested } from 'class-validator';
+import { IdField } from '../../../common';
+import { Post } from './post.dto';
+
+@InputType()
+export class CreatePost {
+  @IdField({ nullable: true })
+  readonly projectId?: string;
+
+  @IdField({ nullable: true })
+  readonly engagementId?: string;
+
+  @IdField({ nullable: true })
+  readonly parentPostId?: string;
```

Comment:
One way I've implemented discussion in the past is to have the parentId take any kind of entity, that way, a starting post can be attached to anything and you don't need to hard code exactly what type the parent is, since we'll be putting discussion on an ever increasing number of things. Something to think about, but not sure I want to slow anything down right now.